src/UnitTests/LanguageServerClient/UnitTests.LanguageServerClient/Mocks/MockLanguageClientBroker.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Python Tools for Visual Studio
+// Copyright(c) Microsoft Corporation
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the License); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED ON AN  *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY
+// IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache Version 2.0 License for specific language governing
+// permissions and limitations under the License.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.LanguageServer.Client;
```

Comment:
code doesn't look like it is using any implementation but it is using interface defined in these namespace? but also, our implementatino doesn't plug into VS as well.

so, if we don't want this VS dependency, we can define the interface our own? rather than reuse the interfaces causing some confusion?

Reply:
ah, it is using the library code such as Connection and MessageHead

Reply:
actually MessageHandler is from JsonRpc (StreamJsonRpc), you don't need to use Connection and use Stream directly since JsonRpc accepts the streams you used to create Connection anyway. so I think we don't need to use LanguageClient for VS for testing.

Reply:
correct. I could duplicate the interfaces but I thought it would be easier to just include that client using.   The plus side of duplicating the interfaces is that I wont have to ignore the build warning complaint about different .net framework

Reply:
@bschnurr 
I didn't mean duplicates. I meant creating simpler version that doesnt have anything that matter to use.

for example, activate, load, onStart all those.. we don't actually need right? we just need 1 like Activate, or Create?

and bunch of other APIs currently just do nothing are not needed as well.